### PR TITLE
Show help info instead of crashing if Android SDK is not found

### LINKED ISSUE #16832: Flutter Crash on Mac without Android SDK
## Steps to Reproduce

Run `flutter run` inside examples/hello_world on a Mac without Android SDK (but with adb and Android devices connected). It would be nice to prompt the information to ask the user to install the Android SDK instead of crashing.

## Logs

```
#0      Object.noSuchMethod (dart:core-patch/dart:core/object_patch.dart:46)
#1      updateLocalProperties (package:flutter_tools/src/android/gradle.dart:212)
#2      _readGradleProject (package:flutter_tools/src/android/gradle.dart:98)
<asynchronous suspension>
#3      _gradleProject (package:flutter_tools/src/android/gradle.dart:90)
<asynchronous suspension>
#4      getGradleAppOut (package:flutter_tools/src/android/gradle.dart:84)
<asynchronous suspension>
#5      AndroidApk.fromCurrentDirectory (package:flutter_tools/src/application_package.dart:88)
<asynchronous suspension>
#6      getApplicationPackageForPlatform (package:flutter_tools/src/application_package.dart:272)
<asynchronous suspension>
#7      FlutterDevice.runHot (package:flutter_tools/src/resident_runner.dart:269)
<asynchronous suspension>
#8      HotRunner.run (package:flutter_tools/src/run_hot.dart:216)
<asynchronous suspension>
#9      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:401)
<asynchronous suspension>
#10     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:333)
<asynchronous suspension>
#11     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:270)
<asynchronous suspension>
#12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142)
<asynchronous suspension>
#13     _rootRun (dart:async/zone.dart:1126)
#14     _CustomZone.run (dart:async/zone.dart:1023)
#15     runZoned (dart:async/zone.dart:1501)
#16     AppContext.run (package:flutter_tools/src/base/context.dart:141)
<asynchronous suspension>
#17     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:261)
#18     CommandRunner.runCommand (package:args/command_runner.dart:194)
<asynchronous suspension>
#19     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:309)
<asynchronous suspension>
#20     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142)
<asynchronous suspension>
#21     _rootRun (dart:async/zone.dart:1126)
#22     _CustomZone.run (dart:async/zone.dart:1023)
#23     runZoned (dart:async/zone.dart:1501)
#24     AppContext.run (package:flutter_tools/src/base/context.dart:141)
<asynchronous suspension>
#25     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:265)
<asynchronous suspension>
#26     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:109)
#27     new Future.sync (dart:async/future.dart:224)
#28     CommandRunner.run (package:args/command_runner.dart:109)
#29     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:174)
#30     run.<anonymous closure> (package:flutter_tools/runner.dart:54)
<asynchronous suspension>
#31     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:142)
<asynchronous suspension>
#32     _rootRun (dart:async/zone.dart:1126)
#33     _CustomZone.run (dart:async/zone.dart:1023)
#34     runZoned (dart:async/zone.dart:1501)
#35     AppContext.run (package:flutter_tools/src/base/context.dart:141)
<asynchronous suspension>
#36     runInContext (package:flutter_tools/src/context_runner.dart:41)
<asynchronous suspension>
#37     run (package:flutter_tools/runner.dart:49)
#38     main (package:flutter_tools/executable.dart:48)
<asynchronous suspension>
#39     main (file:///Users/liyuqian/flutter/flutter/packages/flutter_tools/bin/flutter_tools.dart:8)
#40     _startIsolate.<anonymous closure> (dart:isolate-patch/dart:isolate/isolate_patch.dart:277)
#41     _RawReceivePortImpl._handleMessage (dart:isolate-patch/dart:isolate/isolate_patch.dart:165)
```

## Flutter Doctor

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v0.3.3-pre.4, on Mac OS X 10.13.3 17D102, locale en-US)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set $ANDROID_HOME to that location.
[!] iOS toolchain - develop for iOS devices (Xcode 9.2)
    ✗ Xcode requires additional components to be installed in order to run.
      Launch Xcode and install additional required components when prompted.
    ✗ libimobiledevice and ideviceinstaller are not installed. To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
        For more info, see https://flutter.io/platform-plugins
      To install:
        brew install cocoapods
        pod setup
-Error executing simctl: -6
dyld: Library not loaded: /Library/Developer/PrivateFrameworks/CoreSimulator.framework/Versions/A/CoreSimulator
  Referenced from: /Applications/Xcode.app/Contents/Developer/usr/bin/simctl
  Reason: image not found

[✓] Android Studio (version 3.0)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] IntelliJ IDEA Community Edition (version 2017.3.5)
[✓] Connected devices (1 available)

! Doctor found issues in 2 categories.
```

### PR DESCRIPTION
**Pull request by liyuqian**
Fixes https://github.com/flutter/flutter/issues/16832

### CODE REVIEW

**Review by tvolkert — COMMENTED** [review]
Test?

**Comment by liyuqian** [comment]
@tvolkert : test added.

**Comment by liyuqian** [comment]
Done.

File: packages/flutter_tools/test/android/gradle_test.dart [hunk 1]
```diff
@@ -2,13 +2,32 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import '../src/common.dart';
```
Comment (tvolkert) [inline packages/flutter_tools/test/android/gradle_test.dart]: Nit: sort the local imports like this after the package imports, in a new paragraph (separated by a newline)

**Review by tvolkert — APPROVED** [review]

**Comment by tvolkert** [comment]
FYI, the bots are still running on the most recent commit.  Please wait for the bots to go green before merging.

**Comment by liyuqian** [comment]
Oh, sorry, I thought it would wait for bots to finish before merging (like
Gerrit). Could we enforce that behavior for github?

**Comment by tvolkert** [comment]
I don't think so

**Pull request merged**